module2-database-technologies/README.md

```diff
@@ -1 +1,372 @@
+    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+)  
+```
+
+We can now add some data using `INSERT INTO`. Note that we do not need to
+manually provide an id or a created timestamp, these are automatically generated
+by MySQL.
+
+```sql
+INSERT INTO customer (first_name, last_name) VALUES ('Halit', 'Batur');
+INSERT INTO customer (first_name, last_name) VALUES ('Ammar', 'Sammour');
+```
+
+Now let's `SELECT` the data -- that is, let's view the data.
+```sql
+SELECT * FROM customer;
+```
+
+The `*` symbol tells us to get all columns. You should see something like this:
+
+```
+ id | first_name | last_name |          created_at
+----+------------+-----------+-------------------------------
+  1 | Halit      | Batur     | 2021-08-29 03:33:14.559122-07
+  2 | Ammar      | Sammour   | 2021-08-29 03:34:08.522222-07
+(1 row)
+```
+
+However, you could select only certain columns by typing the name of the column:
+```sql
+SELECT first_name FROM customer;
+```
+
+Finally, let's create a table with a foreign key, the `order` table, noting the
+syntax that is used to create a foreign key that references our `customer`
+table.
+
+```sql
+CREATE TABLE student (
+  id INT AUTO_INCREMENT PRIMARY KEY,
+  product VARCHAR(255) NOT NULL,
+  delivered BOOL,
+  customer_id INT FOREIGN KEY REFERENCES customer(id)
+);
+```
+
+### Understanding joins
+
+**Join** refers to the process of combining two tables together based on a
+common column. In this case, for example, if we want to answer, in one query,
+the question: what did Ammar order? We would write a query like so:
+
+```
+SELECT product FROM order INNER JOIN customers on orders.customer_id =
+customer.id WHERE customer.id = 2;
+```
+
+On your own, try running only the first part of this query and see what happens.
+In your own words, what does this give you?
+
+```
+SELECT product FROM order INNER JOIN customers on orders.customer_id;
+```
+
+You can read more about [SQL
+joins](https://www.educative.io/blog/what-are-sql-joins) here. There are many
+types of joins, but an inner join will by far the most common one.
+
+
+### Further resources for SQL
+* [This repository](https://github.com/LambdaSchool/Relational-Databases) provides
+a good additional resource for reviewing SQL databases and some common parts of
+the query language.
+* When in doubt, you can simply Google, for example, "how to select in MySQL" and check
+    StackOverflow.
+
+
+### SQL queries practice
+Now try to run some SQL queries against the tables that you created:
+* Create at least two new customers
+* Create at least three orders for each customer
+* Select the last names of all customers
+* Select the users, sorted by oldest first (hint: use Google)
+* Use the `LIMIT` keyword to select only one order
+* *Optional:* write a query that returns two columns: the ID of each user and
+    the number of orders for that user. Hint: you need join, use `GROUP BY`, and
+    use `COUNT(*)`. Use Google to read about these.
```

Comment:
This section can be removed, it will go into the SQL queries practice assignment repo instead. (This monorepo includes only readings)